omigami_client/spec2vec.py

```diff
@@ -37,18 +39,34 @@ def match_spectra_from_path(self, mgf_path: str, n_best: int) -> List[pd.DataFra
         A list of pandas dataframes containing the best matches.
 
         """
+
         spectra_generator = load_from_mgf(mgf_path)
-        payload = self._build_payload(spectra_generator, n_best)
-        api_request = self._send_request(payload)
-        prediction = self._format_results(api_request)
 
-        return prediction
+        # issue requests respecting the spectra limit per request
+        predictions = []
+        while True:
+            payload, done = self._build_payload(spectra_generator, n_best)
+            if done:
+                break
+
+            api_request = self._send_request(payload)
+            predictions.extend(self._format_results(api_request))
+        return predictions
 
     def _build_payload(
         self, spectra_generator: Generator[Spectrum, None, None], n_best_spectra: int
-    ) -> Payload:
-        """Extract abundance pairs and Precursor_MZ data, then build the json payload"""
+    ) -> (JSON, bool):
+        """Extract abundance pairs and Precursor_MZ data, then build the json payload
+
+        Returns:
+        --------
+        - payload: JSON
+            the full request payload with input data
+        - finished: bool
+            if we are done issuing requests (all spectra from generator was processed)
```

Comment:
I still dont like this function returning a tuple. Even the description you added here shows you there is something wrong: you have references to issuing requests in a method that should be just building payloads.